tests/kani/write_build_ninja.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import re
+import subprocess
+import sys
+
+import ninja_syntax
+
+RULE_NAME = 'kani'
+BUILD_COUNT = 0
+
+def rust_sources(root):
+    sources = subprocess.run(
+        ['find', root, '-name', '*.rs'],
+        encoding='utf-8',
+        universal_newlines=True,
+        check=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    ).stdout.splitlines()
+    # source files with names including 'ignore' and 'fixme' are broken
+    # source file 'ForeignItems/main.rs' is also broken
+    tags = ['ignore', 'fixme', 'tests/kani/ForeignItems/main.rs']
+    return [src for src in sources if all(tag not in src for tag in tags)]
+
+def kani_pragma(pragma, rust_code):
+    # Examples of kani pragmas:
+    # // kani-verify-fail
+    # // kani-check-fail
+    # // kani-flags: --default-unwind 8
+    try:
+        regexp = fr'//[ \t]*{pragma}([^\n]*)'
```

Comment:
`\s*` rather than `[ \t]*`?

Reply:
No, `\s` will match against a newline `\n` and I want to match to end of the "line".